File: reference/rfid.sdk.win.V02.09/rfid_reader/HF/samples/delphi/TagAccess/rfidlib_aip_iso14443a.pas
unit rfidlib_aip_iso14443a;

{
  only for WIN32
}

interface
type
PBYTE=^byte  ;

function  ISO14443A_GetLibVersion( buf:pchar; nSize:longword):longword;stdcall ;external 'rfidlib_aip_iso14443a.dll' ;
function  ISO14443A_CreateInvenParam( hInvenParamSpecList:longword ; AntennaID:byte):longword; stdcall ;external 'rfidlib_aip_iso14443a.dll' ;
function  ISO14443A_ParseTagDataReport( hTagReport:longword ; var aip_id:longword ; var tag_id:longword ; var ant_id:longword ;  uid:PBYTE; var uidlen:byte ):integer; stdcall ;external 'rfidlib_aip_iso14443a.dll' ;

function MFCL_Connect( hr :longword ;tagType:byte;uid:PBYTE ;var ht:longword):integer ;stdcall ; external 'rfidlib_aip_iso14443a.dll' ;
function MFCL_Authenticate(hr:longword;ht:longword;blkAddr:Byte ; keyType:Byte ;key:PBYTE)  :integer;stdcall ; external 'rfidlib_aip_iso14443a.dll' ;
function MFCL_ReadBlock(hr:longword ;ht:longword; blkAddr:Byte ;blkData:PBYTE ; nSize:longword) :integer ; stdcall ;external 'rfidlib_aip_iso14443a.dll' ;
function MFCL_WriteBlock(hr:longword  ; ht:longword ;blkAddr:Byte ;blkData:PBYTE) :integer ; stdcall ;external 'rfidlib_aip_iso14443a.dll' ;


implementation


end.

File: reference/rfid.sdk.win.V02.09/rfid_reader/HF/samples/VB.net/TagAccess/rfidlib_aip_iso14443A.vb
Namespace RFIDLIB
    Class rfidlib_aip_iso14443A
        Public Declare Function ISO14443A_CreateInvenParam Lib "rfidlib_aip_iso14443A.dll" (ByVal hInvenParamSpecList As UIntPtr, ByVal AntennaID As Byte) As Integer

        Public Declare Function ISO14443A_ParseTagDataReport Lib "rfidlib_aip_iso14443A.dll" (ByVal hTagReport As UIntPtr, ByRef aip_id As UInt32, ByRef tag_id As UInt32, ByRef ant_id As UInt32, ByVal uid As Byte(), ByRef uidlen As Byte) As Integer

        Public Declare Function NTAG21x_Connect Lib "rfidlib_aip_iso14443A.dll" (ByVal hr As UIntPtr, ByVal uid As Byte(), ByRef ht As UIntPtr)

    End Class
End Namespace